src/intrinsics/prepack/utils.js

```diff
@@ -111,3 +112,22 @@ export function createAbstract(
     result = AbstractValue.createAbstractConcreteUnion(realm, result, ...additionalValues);
   return result;
 }
+
+export function createAbstractObject(realm: Realm, name: string, template?: ObjectValue): AbstractObjectValue {
```

Comment:
We have a rule that all abstract values MUST be created in AbstractValue.

Reply:
I can move this function out. Thanks for letting me know.